lib-api/src/main/java/org/ergoplatform/appkit/BoxOperations.java

```diff
@@ -63,8 +61,9 @@ public static BoxOperations createForSenders(List<Address> senders, BlockchainCo
      */
     public static BoxOperations createForEip3Prover(ErgoProver senderProver, BlockchainContext ctx) {
         List<Address> eip3Addresses = senderProver.getEip3Addresses();
-        checkState(eip3Addresses.size() > 0,
-            "EIP-3 addresses are not derived in the prover (use ErgoProverBuilder.withEip3Secret)");
+        if (eip3Addresses.isEmpty()) {
+            throw new IllegalStateException("EIP-3 addresses are not derived in the prover (use ErgoProverBuilder.withEip3Secret)");
+        }
```

Comment:
Why not add checkState to the InternalUtil?

Reply:
Felt unnecessary for 2 uses, and in my opinion this code is clearer. But I can add it if you want. @aslesarenko